tinygrad/tensor.py

```diff
@@ -502,9 +502,8 @@ def softplus(self, beta=1): return (1/beta) * (1 + (self*beta).exp()).log()
   def softsign(self): return self / (1 + self.abs())
 
   # ***** broadcasted binary mlops *****
-
   def _broadcasted(self, fxn:Type[Function], other:Union[Tensor, float], reverse:bool=False) -> Tensor:
-    x,y = [Tensor(t, device=self.device, requires_grad=False) if not isinstance(t, Tensor) else t for t in ([other,self] if reverse else [self,other])]
+    x,y = [Tensor(t, device=self.device, requires_grad=False, dtype=self.dtype if self.dtype != dtypes.bool else dtypes.float32) if not isinstance(t, Tensor) else t for t in ([other,self] if reverse else [self,other])]
```

Comment:
Hmm, why the special check for bool?

Reply:
Without it, some tests fail with the following error:

`RuntimeError: Subtraction, the - operator, with a bool tensor is not supported. If you are trying to invert a mask, use the ~ or logical_not() operator instead.` this is thrown by Pytorch after Tinygrad parse and sends the onnx model

like this one `OnnxBackendNodeModelTest::test_selu_default_expanded_ver18_cpu` because of that, I left the boolean tensors defaulting to float32 like there were before my commit.